src/FlowAware.cpp

```diff
@@ -211,6 +201,149 @@ void IR2Vec_FA::generateFlowAwareEncodings(std::ostream *o,
   if (cyclicCount)
     *cyclicCount << (M.getSourceFileName() + "\t" +
                      std::to_string(cyclicCounter) + "\n");
+  // clock_t end = clock();
+  // double elapsed = double(end - start) / CLOCKS_PER_SEC;
+  // printf("Time measured: %.6f seconds.\n", elapsed);
+}
+
+// newly added
+// This function will update funcVecMap by doing DFS starting from parent
+// function
+void IR2Vec_FA::updateFuncVecMap(
+    llvm::Function *function,
+    llvm::SmallMapVector<const llvm::Function *,
+                         llvm::SmallVector<const llvm::Function *, 10>, 16>
+        &funcCallMap,
+    std::unordered_set<const llvm::Function *> &visitedFunctions) {
+  visitedFunctions.insert(function);
+  SmallVector<Function *, 15> funcStack;
+  funcStack.clear();
+  auto tmpParent = func2Vec(*function, funcStack);
+  // funcVecMap is updated with vectors returned by func2Vec
+  funcVecMap[function] = tmpParent;
+  auto calledFunctions = funcCallMap[function];
+  for (auto &calledFunction : calledFunctions) {
+    if (!calledFunction->isDeclaration() &&
+        visitedFunctions.count(calledFunction) == 0) {
+      // doing casting since calledFunctionsis of type of const llvm::Function*
+      // and we need llvm::Function* as argument
+      Function *foo = const_cast<Function *>(calledFunction);
+      // errs() << foo->getName().str() << "\n";
+      // This function is called recursively to update funcVecMap
+      updateFuncVecMap(foo, funcCallMap, visitedFunctions);
+    }
+  }
+}
+
+void IR2Vec_FA::generateFlowAwareEncodingsForFunction(
+    std::ostream *o, std::string name, std::ostream *missCount,
+    std::ostream *cyclicCount) {
+  // clock_t start = clock();
+  collectWriteDefsMap(M);
+  int noOfFunc = 0;
+  for (auto &f : M) {
+    auto funcName = f.getName().str();
+    // getting demangled function name
+    std::size_t sz = 17;
+    int status;
+    char *const readable_name =
```

Comment:
Please create a function to get demangled method name and call it in both the places.

Reply:
Also appropriately change the existing logic in `generateFlowAwareEncodings()` to make the call to the same function.